### ISSUE TITLE
Apply different sample weighting to individual outputs

### ISSUE THREAD

**Issue by LarsHH** [fchollet/keras#4151]
Hi all,

I am working on a model that does a binary classification and a regression jointly. The idea is that I only want to do the regression when the classification is positive. 
So during training I have the classification and the regression output, but then for the regression I'd like to ignore those samples where the classification target is a zero.

I have considered sample_weighting but this applies to the entire output.

Another option would be to write a custom mse loss that ignores nan's and then put nan's as regression targets for 0-classification targets. But then I will also ignore nan's in the loss if they are there due to an error, so I'd like to avoid that.

Are there other ways I can achieve what I'm trying to do ?

Thank you!
